config/http_config.go

```diff
@@ -210,6 +210,9 @@ func (c *HTTPClientConfig) Validate() error {
 		if c.BasicAuth != nil {
 			return fmt.Errorf("at most one of basic_auth, oauth2 & authorization must be configured")
 		}
+		if len(c.OAuth2.ClientID) < 1 || (len(c.OAuth2.ClientSecret) < 1 || len(c.OAuth2.ClientSecretFile) < 1) {
```

Comment:
We generally use == 0 to check this.

Can we have 2 distincts errors? This would be more readable and more direct for the user.

Reply:
We also need the token url mandatory.